knative-operator/pkg/common/deployment_test.go

```diff
@@ -51,20 +51,44 @@ func TestUpdateWithInvalidController(t *testing.T) {
 	}
 }
 
-func TestUpdateWithPodSpec(t *testing.T) {
-	os.Setenv("HTTP_PROXY", httpProxy)
+func TestProxySettingForHTTPProxy(t *testing.T) {
+	os.Setenv("HTTP_PROXY", proxyValue)
+	client := fake.NewFakeClient(
+		mockController(appsv1.DeploymentSpec{
+			Replicas: ptr.Int32(1),
+			Template: v1.PodTemplateSpec{
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{{
+						Env: []v1.EnvVar{{
+							Name:  "HTTP_PROXY",
+							Value: proxyValue,
+						}},
+					}},
+				},
+			},
+		}, deploymentName))
+	ks := &servingv1alpha1.KnativeServing{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      servingName,
+			Namespace: namespace,
+		},
+	}
+	if err := common.ApplyProxySettings(ks, client); err != nil && apierrors.IsNotFound(err) {
```

Comment:
Why are we only looking at IsNotFound errors? I feel like all errors should cause a failure, no? THis applies to all tests below too.

Reply:
Yep that was a mistake i don't know why i have added condition like that as i have handled [here](https://github.com/openshift-knative/serverless-operator/blob/master/knative-operator/pkg/common/deployment.go#L22-L24) my intention was if at all particular deployment not found i should not continue and return nil but other cases return error